src/ReverseProxy/Service/Proxy/HttpProxy.cs

```diff
@@ -304,6 +304,28 @@ public HttpProxy(ILogger<HttpProxy> logger, IClock clock)
             return (destinationRequest, requestContent);
         }
 
+        private static void RestoreUpgradeHeaders(HttpContext context, HttpRequestMessage destinationRequest, bool isUpgradeRequest)
+        {
+            if (isUpgradeRequest && context.Request.Headers.TryGetValue(HeaderNames.Connection, out var connectionValue))
+            {
+                var hasConnectionUpgrade = false;
+                foreach (var headerValue in connectionValue)
+                {
+                    if (headerValue.Contains("upgrade"))
+                    {
+                        hasConnectionUpgrade = true;
+                        break;
+                    }
+                }
```

Comment:
We want a more precise match on the upgrade value, and to preserve the original input whenever possible.
```suggestion
            if (isUpgradeRequest)
            {
                var connectionValues =  context.Request.Headers.GetCommaSeperatedValues(HeaderNames.Connection);
                string connectionUpgradeValue = null;
                foreach (var headerValue in connectionValues)
                {
                    if (headerValue.Equals("upgrade", StringComparison.OrdinalIgnoreCase))
                    {
                        connectionUpgradeValue = headerValue;
                        break;
                    }
                }
```

Reply:
Done